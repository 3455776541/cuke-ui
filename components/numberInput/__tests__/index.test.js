import React from "react";
import { render, shallow } from "enzyme";
import toJson from "enzyme-to-json";
import NumberInput from "../index";

describe("<NumberInput/>", () => {
  it("should render NumberInput", () => {
    const wrapper = render(
      <div>
        <NumberInput />
      </div>
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("should render addon with input", () => {
    const wrapper = render(
      <div>
        <NumberInput addonBefore={"www"} placeholder="请输入" />
        <NumberInput addonAfter={".com"} placeholder="填写网址" />
        <NumberInput
          addonBefore={"https://"}
          addonAfter={".cn"}
          placeholder="www.lijinke"
        />
      </div>
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("should emit onChange events", () => {
    const onChange = jest.fn();
    const wrapper = shallow(
      <div>
        <NumberInput onChange={onChange} type="text" />
      </div>
    );
    setTimeout(() => {
      wrapper.find("input").simulate("change");
      expect(onChange).toHaveBeenCalled();
    }, 20);
  });
});
