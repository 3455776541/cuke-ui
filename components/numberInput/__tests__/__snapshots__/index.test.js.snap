// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NumberInput/> should render NumberInput 1`] = `
<div>
  <input
    class="cuke-input cuke-number-input"
    placeholder=""
    type="text"
  />
</div>
`;

exports[`<NumberInput/> should render addon with input 1`] = `
<div>
  <span
    class="cuke-number-input-group cuke-number-input-group-addon-before"
  >
    <span
      class="cuke-number-input-group-addon"
    >
      www
    </span>
    <input
      class="cuke-input cuke-number-input"
      placeholder="请输入"
      type="text"
    />
  </span>
  <span
    class="cuke-number-input-group cuke-number-input-group-addon-after"
  >
    <input
      class="cuke-input cuke-number-input"
      placeholder="填写网址"
      type="text"
    />
    <span
      class="cuke-number-input-group-addon"
    >
      .com
    </span>
  </span>
  <span
    class="cuke-number-input-group cuke-number-input-group-addon-before cuke-number-input-group-addon-after cuke-number-input-group-addon-all"
  >
    <span
      class="cuke-number-input-group-addon"
    >
      https://
    </span>
    <input
      class="cuke-input cuke-number-input"
      placeholder="www.lijinke"
      type="text"
    />
    <span
      class="cuke-number-input-group-addon"
    >
      .cn
    </span>
  </span>
</div>
`;
